crypto/aes.go

```diff
@@ -0,0 +1,58 @@
+package crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"io"
+)
+
+func AESEncrypt(key []byte, message string) (string, error) {
+	c, err := aes.NewCipher(key)
+	if err != nil {
+		return "", err
+	}
+
+	gcm, err := cipher.NewGCM(c)
+	if err != nil {
+		return "", err
+	}
+
+	nonce := make([]byte, gcm.NonceSize())
+	if _, err = io.ReadFull(rand.Reader, nonce); err != nil {
+		return "", err
+	}
+
+	plaintext := []byte(message)
+	ciphertext := gcm.Seal(nonce, nonce, plaintext, nil)
```

Comment:
not very familiar with the package, but from what I read, isn't the destination supposed to be nil? `gcm.Seal(nil, nonce, plaintext, nil)`

Reply:
Please check this, it will be stored along with the ciphertext in terms of `authentication tag`.
https://stackoverflow.com/questions/68350301/extract-tag-from-cipher-aes-256-gcm-golang

Reply:
I don't get it, how does it relate with the SO article? 

I was reading this https://pkg.go.dev/crypto/cipher#:~:text=//%20Seal%20encrypts%20and,byte)%20%5B%5Dbyte
and it seems that the first parameter is just a variable to write the result, i may be totally wrong though :slightly_smiling_face:

Reply:
And I referred to this: https://www.melvinvivas.com/how-to-encrypt-and-decrypt-data-using-aes